tensorflow/python/keras/applications/efficientnet_weight_update_util.py

```diff
@@ -0,0 +1,307 @@
+"""Utils for EfficientNet models for Keras.
+Write weights from  ckpt file as in original repo
+(https://github.com/tensorflow/tpu/tree/master/models/official/efficientnet)
+to h5 file for keras implementation of the models.
+"""
+
+import tensorflow as tf
+import tensorflow.keras as keras
+import h5py
+import numpy as np
+import argparse
+from tensorflow.keras.applications.efficientnet import *
+
+def write_ckpt_to_h5(path_h5, path_ckpt, keras_model, use_ema=True):
+  """ Map the weights in checkpoint file (tf) to h5 file (keras)
+  
+  Args:
+    path_h5: str, path to output hdf5 file to write weights loaded
+      from ckpt files.
+    path_ckpt: str, path to the ckpt files (e.g. 'efficientnet-b0/model.ckpt')
+      that records efficientnet weights from original repo 
+      https://github.com/tensorflow/tpu/tree/master/models/official/efficientnet
+    keras_model: keras model, built from keras.applications efficientnet
+      functions (e.g. EfficientNetB0)
+    use_ema: Bool, whether to use ExponentialMovingAverage result or not
+  """
+  model_name_keras = keras_model.name
+  model_name_tf = model_name_keras_to_tf(model_name_keras)
+  keras_model.save_weights(path_h5)
+
+  keras_weights = get_h5_names(path_h5)
+  tf_weights = get_tf_names(path_ckpt)
+  blocks_keras = get_keras_blocks(keras_weights)
+  blocks_tf = get_tf_blocks(tf_weights)
+
+  with tf.compat.v1.Session() as sess:
+    saver = tf.compat.v1.train.import_meta_graph(f'{path_ckpt}.meta')
+    saver.restore(sess, path_ckpt)
+    graph = tf.compat.v1.get_default_graph()
+    v_all = tf.compat.v1.global_variables()
+
+    for keras_block, tf_block in zip(blocks_keras, blocks_tf):
+      print(f'working on block {keras_block}, {tf_block}')
+      for keras_name in keras_weights:
+        if keras_block in keras_name:
+          tf_name = keras_name_to_tf_name_block(keras_name, keras_block=keras_block, tf_block=tf_block, use_ema=use_ema, model_name_tf=model_name_tf)
+          for v in v_all:
+            if v.name == tf_name:
+              v_val = sess.run(v)
+              with h5py.File(path_h5, 'a') as f:
+                v_prev = f[keras_name][()]
+                f[keras_name].write_direct(v_val)
+              print(f'writing from: {tf_name}\n  to: {keras_name}')
+              print(f'  average change: {abs(v_prev - v_val).mean()}')
+              v_all.remove(v)
+              break
```

Comment:
Why break here? should we convert all the v from v_all?

Reply:
eg, i see. If you are trying to find the v in v_all with the same name, how about [v for v in v_all if v.name ==tf_name]?